system/Database/Forge.php

```diff
@@ -289,12 +267,109 @@ public function addKey($key, $primary = false)
 		{
 			$this->keys[] = $key;
 		}
-
 		return $this;
 	}
-
+	
 	//--------------------------------------------------------------------
-
+	
+	/**
+	 * Add foreign Key
+	 * 
+	 * @param string $key       Database field to reference.
+	 * @param string $fkey      Foreign key field.
+	 * @param string $ftable    Foreign table to reference.
+	 * @param string $onUpdate  On update event.
+	 * @param string $onDelete  On delete event.
+	 */
+	public function addForeignKey(string $key, string $fkey, string $table, string $ftable,
+	        string $onUpdate = 'CASCADE', string $onDelete = 'CASCADE')
+	{
+	    $this->foreignKeys[] = [
+	            'key' => $key,
+	            'fkey' => $fkey,
+	            'table' => $table,
+	            'ftable' => $ftable,
+	            'onUpdate' => $onUpdate,
+	            'onDelete' => $onDelete
+	    ];
+	    
+	    // If not already an index, add key
+	    if ( ! in_array($key, $this->keys))
+	    {
+	        $this->keys[] = $key;
+	    }
+	    
+	    return $this;
+	}
+	
+	//--------------------------------------------------------------------
+	
+	/**
+	 * Get foreign Keys
+	 * 
+	 */
+	public function getForeingKeys()
+	{
+	    return $this->foreignKeys;
+	}
+	
+	//--------------------------------------------------------------------
+	
+	/**
+	 * Creates foreign keys.
+	 * 
+	 * @param string $table
+	 */
+	public function createForeignKeys()
+	{
+	    foreach ($this->foreignKeys as $fk)
+	    {
+    	    $this->db->query($this->compileFKeyCreate($fk));
+	    }
+	}
+	
+	//-------------------------------------------------------------------
+	
+	/**
```

Comment:
Docblocks should always contain at least a basic description of what the method does.